src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -128,7 +134,7 @@ public JsonClassInfo(Type type, JsonSerializerOptions options)
                                             // Overwrite the one just added since it has [JsonIgnore].
                                             cache[jsonPropertyInfo.NameAsString] = jsonPropertyInfo;
                                         }
-                                        else if (other.PropertyInfo?.Name != jsonPropertyInfo.PropertyInfo?.Name &&
+                                        else if (other.MemberInfo?.Name != jsonPropertyInfo.MemberInfo?.Name &&
```

Comment:
FWI you'll likely have to react to https://github.com/dotnet/runtime/pull/36936 here.